src/brod_utils.erl

```diff
@@ -210,9 +212,55 @@ fetch(SockPid, ReqFun, MaxBytes, Offset) ->
       end
   end.
 
+%% @doc Get sasl options from client config.
+-spec get_sasl_opt(client_config()) -> sasl_opt().
+get_sasl_opt(Config) ->
+  case proplists:get_value(sasl, Config) of
+    {plain, User, PassFun} when is_function(PassFun) ->
+      {plain, User, PassFun()};
+    {plain, File} ->
```

Comment:
get_sasl_opt and init_sasl_opt are mostly identical. Perhaps leave just this one?